### PR TITLE
DlgPrefSound: fix setting keylock engine from combobox

### PR DESCRIPTION
**Pull request by ronso0**
previously the new engine wasn't applied due to wrong type conversion :grimacing:

### CODE REVIEW

File: src/preferences/dialog/dlgprefsound.cpp [hunk 1]
```diff
@@ -312,9 +312,11 @@ void DlgPrefSound::slotApply() {
     SoundDeviceStatus status = SoundDeviceStatus::Ok;
     {
         ScopedWaitCursor cursor;
-        m_pKeylockEngine->set(keylockComboBox->currentData().toDouble());
+        const auto keylockEngine =
+                keylockComboBox->currentData().value<EngineBuffer::KeylockEngine>();
+        m_pKeylockEngine->set(static_cast<double>(static_cast<double>(keylockEngine)));
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefsound.cpp:317]:
```suggestion
        m_pKeylockEngine->set(static_cast<double>(keylockEngine));
```

**Review by daschuer — COMMENTED** [review]
I do not understand the problem that is fixed here ...

**Review by Swiftb0y — COMMENTED** [review]
The changes seem like a noop to me too. Are you sure this is fixing the problem?

**Comment by Swiftb0y** [comment]
Thanks for looking into it btw.

**Comment by ronso0** [comment]
> I do not understand the problem that is fixed here ...

https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Rubberband.203.2E0.20setting.20not.20applied.2Fsaved
The value from mixxx.cfg is loaded correctly, but no matter what new value you select in the preferences, in slotApply() `keylockComboBox->currentData().toDouble()` would always return 0 and you'd be stuck with SoundTouch.

**Comment by ronso0** [comment]
> The changes seem like a noop to me too. Are you sure this is fixing the problem?

Nope.
Yep.

File: src/preferences/dialog/dlgprefsound.cpp [hunk 2]
```diff
@@ -312,9 +312,11 @@ void DlgPrefSound::slotApply() {
     SoundDeviceStatus status = SoundDeviceStatus::Ok;
     {
         ScopedWaitCursor cursor;
-        m_pKeylockEngine->set(keylockComboBox->currentData().toDouble());
+        const auto keylockEngine =
+                keylockComboBox->currentData().value<EngineBuffer::KeylockEngine>();
+        m_pKeylockEngine->set(static_cast<double>(keylockEngine));
```
Comment (Swiftb0y) [inline src/preferences/dialog/dlgprefsound.cpp:317]: interesting, apparently `QVariant::toDouble` doesn't work if the contained type is an enum? https://doc.qt.io/qt-5/qvariant.html#toDouble 

Am I reading that correctly?

**Review by Swiftb0y — APPROVED** [review]
LGTM, The broken `QVariant<enum_t>::toDouble` is a bad tripping hazard though, we might want to file a QTBUG.

**Review by daschuer — APPROVED** [review]
Thank you for the fix and your fast responds

**Pull request merged**